### DISCUSSION TITLE
How do I delete certain backups from S3 bucket?

### DISCUSSION THREAD

**Discussion by jargals2** [vmware-tanzu/velero#3398] [Community support Q&A]
Does deleting a backup also automatically delete backups stored in S3? I used Restic. It's been over 2 hours since I deleted a backup and my bucket size is still same as before.   If it's not automatically deleted how do I know which files to delete in my bucket? Thank you

**Comment by carlisia** [comment]
Hi @jargals2. Seems something is going wrong. Yes, deleting a backup deletes its files in the S3 bucket. Maybe there was a connection failure that precluded that operation. Maybe it was something else. What do you get when you run `velero get backup`, does your backup appears listed? Also, we'd need to look at the Velero log to understand what happened.

To answer your question, your backup will be a folder named the same as the backup name in the S3 bucket under a folder named `backup`.

**Reply by tariq1278** [reply]
Yes there is folder name with backup..that one deleted but there is another folder name with restic that one not deleted when we triggered velero delete backup command

**Reply by jargals2** [reply]
Same as @tariq1278 . When I delete the backup, corresponding files in the backup directory in S3 is deleted, but corresponding restic backup is not being deleted.

**Reply by dsu-igeek** [reply]
Restic works by storing file data referenced by a SHA-1 hash and then refers to the hashes in a snapshot.  This also provides basic data deduplication and sharing of data across snapshots (https://restic.readthedocs.io/en/stable/100_references.html#design)  Deleting a backup will not necessarily result in any data being deleted from the S3 bucket.  If you take backup 1, for example, and then immediately take backup 2, backup 2 should not result in much additional storage since it is mostly the data fro backup 1.  If you then delete 1 or 2, you won't see much space returned in the S3 bucket because the data is shared across the two backups.

**Reply by jargals2** [reply]
@dsu-igeek Hi, it is behaving the same when there was only one backup in the backup, and deleting it does not delete the corresponding restic backup. I have tried installing different versions of aws plugin and it behaves the same

**Reply by zubron** [reply]
@jargals2 The restic repository controller will not delete unreferenced snapshots upon backup deletion but will instead run this process periodically by running `restic prune`. The default maintenance frequency is once a week: https://github.com/vmware-tanzu/velero/blob/main/pkg/restic/common.go#L48-L50.

You can modify how frequently this maintenance happens by using the `--default-restic-prune-frequency` flag when running `velero install`.

**Reply by tariq1278** [reply]
Restic command is not taking in shell, do we need to any restic plugin install?

**Reply by vitobotta** [reply]
@zubron Hi, same problem here. In /restic I have subfolders for backups deleted a long time ago. Can I safely delete those subfolder or will it screw up something? I also have many subfolders inn /restores because I have used this bucket with Velero for a while and have done many restores. Can I delete those too? Thanks

**Reply by tariq1278** [reply]
How do we get this restic data belog to this backup job, I have seen more than 100+ folder inside data folder under restic folder

**Reply by zubron** [reply]
@vitobotta It's not always clear whether restic data can be deleted as it performs [deduplication of data](https://restic.readthedocs.io/en/latest/100_references.html?highlight=deduplication#backups-and-deduplication).

So although backups have been deleted, the snapshots and blobs in the restic directories may be associated with other backups that you still rely on.

**Reply by zubron** [reply]
@tariq1278 Yes, it's expected to have many directories under the parent restic directory. See their docs on the [repository format](https://restic.readthedocs.io/en/latest/100_references.html?highlight=deduplication#repository-format) for more info. As described in the comment above, snapshots and blobs may belong to multiple backups if the data has been deduplicated.

**Reply by vitobotta** [reply]
> @vitobotta It's not always clear whether restic data can be deleted as it performs [deduplication of data](https://restic.readthedocs.io/en/latest/100_references.html?highlight=deduplication#backups-and-deduplication).
> 
> So although backups have been deleted, the snapshots and blobs in the restic directories may be associated with other backups that you still rely on.

Hi @zubron , some restic folders are old and refer to namespaces I have deleted long time ago, so it's unlikely they are still linked to some existing backups.
